### PR TITLE
chore(data model): drop PartialOrd from Event

### PR DESCRIPTION
**Pull request by lukesteensen**
This reduces the effective API surface and makes it easier to replace
BTreeMap with a more efficient representation.

Signed-off-by: Luke Steensen <luke.steensen@gmail.com>

### CODE REVIEW

**Review by blt — APPROVED** [review]
Very cool. I expect some test dings but that should be a small fix.

**Comment by blt** [comment]
> Just noting we switched to `BTreeMap` in #1838 for its ordering characteristics. I want to make sure we are OK with the behavior change before switching the representation.

I think this is different? The `PartialOrd` didn't ensure that keys ended up ordered in serialized output, more that we could use `<` etc on a `Value` directly. I'm not actually sure whether we have a separate ordering constraint on the keys of `Value` maps or not. I _think_ not, which might be something we ought to resolve.

**Comment by jszwedko** [comment]
> > Just noting we switched to `BTreeMap` in #1838 for its ordering characteristics. I want to make sure we are OK with the behavior change before switching the representation.
> 
> I think this is different? The `PartialOrd` didn't ensure that keys ended up ordered in serialized output, more that we could use `<` etc on a `Value` directly. I'm not actually sure whether we have a separate ordering constraint on the keys of `Value` maps or not. I _think_ not, which might be something we ought to resolve.

Yeah, that is a good point. I just want to make sure we pause and consider the implications before swapping out `BTreeMap` per:

> and makes it easier to replace BTreeMap with a more efficient representation.

from the PR description.

**Comment by jszwedko** [comment]
@lukesteensen is this still relevant? We need to support the sorting behavior for map values (or make a decision not to), but otherwise dropping this seems fine.

**Comment by lukesteensen** [comment]
@jszwedko yes I think this is still desirable.

**Review by jszwedko — APPROVED** [review]

**Comment by jszwedko** [comment]
Looks like clippy is failing, otherwise 👍

File: lib/vector-core/src/event/metric/value.rs [hunk 1]
```diff
@@ -420,6 +420,14 @@ impl PartialEq for MetricValue {
     }
 }
 
+fn float_eq(l_value: f64, r_value: f64) -> bool {
+    if l_value.is_nan() && r_value.is_nan() {
+        true
+    } else {
+        l_value.eq_ulps(&r_value, &1)
+    }
```
Comment (bruceg) [inline lib/vector-core/src/event/metric/value.rs:428]:
```suggestion
    (l_value.is_nan() && r_value.is_nan()) || l_value.eq_ulps(&r_value, &1)
```

**Review by bruceg — APPROVED** [review]

**Pull request merged**